core/src/syn/parser/idiom.rs

```diff
@@ -252,10 +255,58 @@ impl Parser<'_> {
 				self.pop_peek();
 				Part::All
 			}
+			t!("{") => {
+				self.pop_peek();
+				self.parse_destructure_part()?
+			}
 			_ => Part::Field(self.next_token_value()?),
 		};
 		Ok(res)
 	}
+	/// Parse the part after the `.{` in an idiom
+	pub fn parse_destructure_part(&mut self) -> ParseResult<Part> {
+		let mut destructured: Vec<DestructurePart> = Vec::new();
+		loop {
+			match self.peek_kind() {
+				t!("}") => {
+					self.pop_peek();
+					break;
+				}
+				_ => {
+					let field: Ident = self.next_token_value()?;
+					let part = match self.peek_kind() {
+						t!(":") => {
+							self.pop_peek();
+							DestructurePart::Aliased(field, self.parse_local_idiom()?)
+						}
+						t!(".") => {
+							self.pop_peek();
+							let found = self.peek_kind();
+							match self.parse_dot_part()? {
+								Part::All => DestructurePart::All(field),
+								Part::Destructure(v) => DestructurePart::Destructure(field, v),
+								_ => {
+									return Err(ParseError::new(
+										ParseErrorKind::Unexpected {
+											found,
+											expected: "a star or a destructuring",
+										},
+										self.last_span(),
+									))
+								}
+							}
+						}
+						_ => DestructurePart::Field(field),
+					};
+
+					self.eat(t!(","));
```

Comment:
This just eats any possible `,` but doesn't check if it is actually present. Which means that syntax like `foo.{a b c d}` (no `,` between the fields) is also allowed. I think we should probably force having `,` between fields.

Reply:
Ah! I thought that this way it would fail if it did not find the comma. What method should I use instead?

Reply:
Addressed in [be1832d](https://github.com/surrealdb/surrealdb/pull/4443/commits/be1832d5c4b9d3897acef249a31d1e3964fd333e)

It now also requires at least one field to be specified in the destructure, which is fine as an empty destructure doesn't make sense anyways

Reply:
It also now doesn't support trailing commas which it probably should since most delimited sequences in surrealql support trailing seperators. The pattern I often use for delimited sequences is first check if the closing delimiter is the next token, quit if so, otherwise parse a value then check for the comma, if there is none require the next token to be the delimiter, otherwise continue the loop.